<!--
-->
<launch>
  
  # Use simulation time for TFs
  <param name="use_sim_time" value="true" /> 

  # Bag file
  <node name="bag_player" exec="play" pkg="ros2bag" args="mbzirc.bag -s 0 -r 1.0 --clock " />

  # Odometry from odom to base_link already computed into the bag file
  # Odometry computed with LOAM

  # mbzirc map conf
  <arg name="base_frame_id" default="base_link"/>
  <arg name="odom_frame_id" default="odom"/>
  <arg name="global_frame_id" default="world"/>
  <arg name="initial_x" default="16.964"/>
  <arg name="initial_y" default="61.508"/>
  <arg name="initial_z" default="1.0"/>
  <arg name="initial_a" default="-0.063"/> 
  <arg name="map" default="mbzirc.bt" />
  <arg name="map_path" default="$(find-pkg-share dll)/maps/$(arg map)"/>

  # Launch DLL
  <node name="dll_node" exec="dll_node" pkg="dll" output="screen">
    <remap from="/dll_node/initial_pose" to="/initialpose"/>
    <remap from="/imu" to="/dji_sdk/imu"/>
    <param name="in_cloud" value="/os1_cloud_node/points_non_dense" />
    <param name="base_frame_id" value="$(arg base_frame_id)" />
    <param name="odom_frame_id" value="$(arg odom_frame_id)" />
    <param name="global_frame_id" value="$(arg global_frame_id)" />

    <param name="rate" value="10.0" />
    <param name="map_path" value="$(arg map_path)" />
    <param name="sensor_dev" value="0.05" />   
    <param name="publish_point_cloud" value="true" />    
    <param name="update_min_d" value="0.01" />
    <param name="update_min_a" value="0.01" /> 
    <param name="update_min_time" value="0.1" />
    <param name="initial_x"   value="$(arg initial_x)"/>
    <param name="initial_y"   value="$(arg initial_y)"/>
    <param name="initial_z"   value="$(arg initial_z)"/>
    <param name="initial_a"   value="$(arg initial_a)"/>
    <param name="use_imu" value="true" />
    <param name="align_method" value="1" />  # 1: DLL, 2: NDT, 3: ICP
   
  </node>

  <node name="rviz" exec="rviz2" pkg="rviz2" args="-d $(find-pkg-share dll)/launch/mbzirc.rviz" output="screen"/>

</launch>	
